src/core/ai.js

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright 2018 The boardgame.io Authors
+ *
+ * Use of this source code is governed by a MIT-style
+ * license that can be found in the LICENSE file or at
+ * https://opensource.org/licenses/MIT.
+ */
+
+/**
+ * AI
+ *
+ * Sets up the necessary information for running bots in a game.
+ *
+ * There are two parts that need to be described:
+ * - What are the next possible moves given a particular game state.
+ * - What is the evaluation function (score) for a particular state.
+ *
+ * A bot might use this information to maximize its score and minimize
+ * other player's scores (i.e. MiniMax and similar algorithms).
+ *
+ * @param {...object} possibleMoves - Function with signature (G, ctx) => moves
```

Comment:
We may have different ways of wanting to expand `possibleMoves`. This function provides one such way.

I think this interface will be cleaner and more extensible if we ask the user to provide:

```
ai: {
   possibleMoves: () => [...]
   score: () => {}
}
```

in raw form (the return value of this function, basically), and then we provide different ways to generate `possibleMoves` and `score` independently.

```
ai: {
   possibleMoves: MoveRange(...),
   score: TaskPriority(...),
}
```

Each of these generator functions can accept whatever it wants as long as it generates the raw form correctly.

Reply:
Ok! That was actually my first idea but I was having a hard time to find a place to add this "MoveRange" or "TaskPriority" function. But I think they can sit on ai.js and be part of the exports = {}. 
I will change this later today (or maybe tomorrow, I am kind of sick)

Reply:
Get better soon!

Reply:
Ok this is done now :)